src/WebSocketHandler.js

```diff
@@ -5,246 +5,60 @@
-    }
-
-    if (this.ws) {
-      const dt = (new Date() - this.latestSetupDate) / 1000;
-      if (dt < this.openConnectionTimeout) {
-        return;
-      }
-      this.ws.close();
-    }
-    this.ws = new this.WebSocket(wsURL);
-    this.latestSetupDate = new Date();
-
-    this.ws.onopen = () => {
-      this.onOpen();
-    }
-    this.ws.onmessage = (evt) => {
-      this.onMessage(evt);
-    }
-    this.ws.onerror = (evt) => {
-      this.onError(evt);
-    }
-    this.ws.onclose = () => {
-      this.onClose();
-    }
-  }
-
-  onPong() {
-    if (this.latestPingDate) {
-      const dt = (new Date() - this.latestPingDate) / 1000;
-      this.latestRTT = dt;
-      this.latestPingDate = null;
-    }
-    if (this.timeoutTimer) {
-      clearTimeout(this.timeoutTimer);
-      this.timeoutTimer = null;
-    }
-  }
-
-  /**
-   * Handle message receiving from websocket
-   *
-   * @param {Object} evt Event that has data (evt.data) sent in the websocket
-   */
-  onMessage(evt) {
-    const message = JSON.parse(evt.data)
-    const _type = message.type.split(':')[0]
-    if (_type === 'pong') {
-      this.onPong();
-    }
-    this.emit(_type, message)
-  }
-
-  /**
-   * Method called when websocket connection is opened
-   */
-  onOpen() {
-    if (this.connected === false) {
-      // If was not connected  we need to reload data
-      // Emits event to reload data
-      this.emit('reload_data');
-    }
-    this.connected = true;
-    this.started = true;
-    this.setIsOnline(true);
-    this.heartbeat = setInterval(() => {
-      this.sendPing();
-    }, this.heartbeatInterval);
-    wallet.onWebsocketOpened();
-  }
-
-  /**
-   * Method called when websocket connection is closed
-   */
-  onClose() {
-    this.started = false;
-    this.connected = false;
-    this.setIsOnline(false);
-    wallet.onWebsocketBeforeClose();
-    if (this.ws) {
-      this.ws.close();
-      this.ws = null;
-    }
+    this.ws = null;
+    /* Right after importing the modules helpers and wallet, they are
+     * not available and are still undefined.
+     * This is probably caused by a cyclic import (wallet -> helpers -> tokens -> wallet)
+     * but still need more study.
+     * For now the setTimeout is used, so we can use the helpers module properly.
+     */
     setTimeout(() => {
-      this.setup()
-    }, this.retryConnectionInterval);
-    clearInterval(this.heartbeat);
-  }
-
-  /**
-   * Method called when an error happend on websocket
-   *
-   * @param {Object} evt Event that contains the error
-   */
-  onError(evt) {
-    this.onClose();
-    // console.log('ws error', window.navigator.onLine, evt);
-  }
-
-  /**
-   * Method called to send a message to the server
-   *
-   * @param {string} msg Message to be sent to the server (usually JSON stringified)
-   */
-  sendMessage(msg) {
-    if (!this.started) {
-      this.setIsOnline(false);
-      return;
-    }
+      this.ws = new WS({ wsURL: helpers.getWSServerURL });
```

Comment:
Missing `()` on function call.

Reply:
The parameter `wsURL` can be either a function or a string.

Reply:
To keep the current behaviour, it must be a function. So, if the server is changed in the storage, it will try to connect to the new server on the next call.

Reply:
Ah, got it. Thanks.